packages/file-pipeline/src/helpers/route-cache.ts

```diff
@@ -0,0 +1,111 @@
+import path from "path"
+import File from "vinyl"
+import {RouteCacheEntry, RouteCacheInterface, RouteType, RouteVerb} from "../types"
+
+export class RouteCache implements RouteCacheInterface {
+  routeCache: Record<string, RouteCacheEntry> = {}
+
+  delete(file: File) {
+    delete this.routeCache[file.originalRelative]
+  }
+
+  normalizePath(input: string) {
+    if (path.sep === path.posix.sep) return input
+    return input.split(path.sep).join(path.posix.sep)
+  }
+
+  getUrifromPath(path: string) {
+    const findStr = "/pages"
+    const findStrIdx = path.indexOf(findStr)
+    const uri = path.substring(findStrIdx + findStr.length, path.lastIndexOf("."))
+    const uriWithoutIndex = uri.replace("/index", "")
+    return uriWithoutIndex.length > 0 ? uriWithoutIndex : "/"
+  }
+
+  getVerb(type: RouteType): RouteVerb {
+    switch (type) {
+      case "api":
+        return "*"
+      case "rpc":
+        return "post"
+      default:
+        return "get"
+    }
+  }
+
+  isErrorCode(uri: string) {
+    if (uri.length === 4) {
```

Comment:
I think it would be nice to save the 4 in a constant and use that here instead. It will make it easier to understand what you are comparing to and why the length should be 4 in this case.

Also, how come you expect the error code to be four vs three? Is there an extra character appended to them in this case?

Reply:
Okay I will make it into constant.

Yes, each uri is prepended by `/` character because of the normalization of URI from function `getUrifromPath`

Reply:
Awesome, thanks for clarifying that